### PR TITLE
Support bluetooth low energy HID Braille displays

### PR DESCRIPTION
**Pull request by alexmoon**
### Summary of the issue:

NVDA does not currently detect standard HID braille displays using the Bluetooth Low Energy transport. This PR adds the BLE GATT HIDS UUID to be detected as a valid Bluetooth HID device.

### Description of user facing changes

BLE HID braille displays will now be detected. No other user facing changes.

### Testing strategy:

Checked that NVDA detects and uses our BLE braille display. Unit/system tests are not applicable due to required hardware interactions.

### Known issues with pull request:

None

### Change log entries:

Bug fixes

- Support Bluetooth Low Energy HID Braille displays

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by codeofdusk** [comment]
Are there any actively used HID displays that communicate over BLE?

**Comment by alexmoon** [comment]
> Are there any actively used HID displays that communicate over BLE?

Yes, Tactile Engineering's Cadence tablet is now making its way onto the market and it uses BLE.

**Comment by josephsl** [comment]
Hi, I see. I would like to suggest that, in the future, when discussing new technology support, let us go through issue+PR pair so we can discuss this in detail on the issue side before coming up with a pull request please. That way we can talk about rationale, supporting documentation (if any(, and have time to discuss specifics and strategies. Thanks.

**Comment by leonardder** [comment]
I recall @bramd has performed some tests with BLE in the past. Would like to know whether he has anything to add.

@josephsl wrote:
> Hi, I see. I would like to suggest that, in the future, when discussing new technology support, let us go through issue+PR pair so we can discuss this in detail on the issue side before coming up with a pull request please. That way we can talk about rationale, supporting documentation (if any(, and have time to discuss specifics and strategies. Thanks.

While this is generally true, I think with a two lines change like this, it makes sense to have the suggestion as part of the pr.

**Comment by seanbudd** [comment]
Is this ready for review?

File: source/hwPortUtils.py [hunk 1]
```diff
@@ -443,6 +443,8 @@ def _getHidInfo(hwId, path):
 		info["usbID"] = hwId[:17] # VID_xxxx&PID_xxxx
 	elif hwId.startswith("{00001124-0000-1000-8000-00805f9b34fb}"):
 		info["provider"] = "bluetooth"
+	elif hwId.startswith("{00001812-0000-1000-8000-00805f9b34fb}"):
```
Comment (seanbudd) [inline source/hwPortUtils.py:446]:
```suggestion
	elif hwId.startswith("{00001812-0000-1000-8000-00805f9b34fb}"):
		# Low energy bluetooth: #15470
```

**Review by seanbudd — APPROVED** [review]
Thanks @alexmoon

**Comment by seanbudd** [comment]
Hi @alexmoon,

We are unable to push changes to your branch.
If possible, please give NV Access or myself push permissions?
[Related docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-permissions-and-visibility-of-forks#about-permissions-of-forks).

Alternatively, could you please integrate the change log entry into `changes.t2t` under feature requests?

```
- Added support for Bluetooth Low Energy HID Braille displays. (#15470)
-

**Comment by alexmoon** [comment]
Thanks @seanbudd!

**Pull request merged**